src/deps_deploy/deps_deploy.clj

```diff
@@ -88,22 +93,81 @@
                  :coordinates (mvn-coordinates coordinates))
   (println "done."))
 
-(defmethod deploy :local [{:keys [artifact-map coordinates]}]
+(defmethod deploy* :local [{:keys [artifact-map coordinates]}]
   (println "Installing" (artifact coordinates) "to your local `.m2`")
   (aether/install :artifact-map artifact-map
                   :transfer-listener :stdout
                   :coordinates (mvn-coordinates coordinates))
   (println "done."))
 
-(defn -main [deploy-or-install artifact & [sign-releases]]
-  (let [pom (slurp "pom.xml")
-        coordinates (coordinates-from-pom pom)
-        versioned-pom (spit (versioned-pom-filename coordinates) pom)]
-
+(defn- tag-val-replacer [tag->val]
```

Comment:
I tend to prefer creating fns which take the required amount of args, and then rather bind the args using `partial` (to which I'm, ehrm, partial) or with an anon-fn `#(replace-tag {version-tag [version] group-id-tag [group-id] ...} %)`

Reply:
👍 Cool.  I'm quite partial to doing things that way too ;-) it was even my first instinct, which I held back on because I figured it'd be bikeshedded the other way.  Happy to change.